### PR TITLE
started maunal doc for swift sdk

### PR DESCRIPTION
**Pull request by bryce-b**

### CODE REVIEW

File: content/en/docs/instrumentation/swift/manual.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
+---
+title: Manual Instrumentation
+linkTitle: Manual
+weight: 5
+description: >-
+<img width="35"
+src="https://raw.github.com/open-telemetry/opentelemetry.io/main/iconography/32x32/Swift.svg"
+alt="Swift"></img>
+Manual instrumentation for opentelemetry-swift
```
Comment (svrnm) [inline content/en/docs/instrumentation/swift/manual.md:9]:
```suggestion
description: Manual instrumentation for opentelemetry-swift
```

File: content/en/docs/instrumentation/swift/manual.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+---
+
+## Setup
+
+The [OpenTelemetrySdk](https://github.com/open-telemetry/opentelemetry-swift/blob/main/Sources/OpenTelemetrySdk/OpenTelemetrySdk.swift#L37) provides limited functionality in its default configuration. For more useful functionality, some configuration is required. 
+
+The default registered `TracerProvider` and `MetricProvider` are not configured with an exported. There are several [exporters](https://github.com/open-telemetry/opentelemetry-swift/tree/main/Sources/Exporters) available depending on your needs. Below we will explore configuring the OTLP exporter, which can be used for sending data to the [opentelemetry-collector](https://github.com/open-telemetry/opentelemetry-collector).
```
Comment (svrnm) [inline content/en/docs/instrumentation/swift/manual.md:16]: Eventually we want to have Exporters in their own page, but I think for a first version of the Swift documentation, this should be ok, @cartermp: what do you think?

File: content/en/docs/instrumentation/swift/manual.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+```
+
+After configuring the MeterProvider & TracerProvider all subsequently initialized instrumentation will be exporting using this OTLP exporter.
+
+## Acquiring a Tracer
+
+To do tracing, you will need a tracer. 
+A tracer is acquired through the tracer provider and is responsible for creating spans. The OpenTelementrySdk manages the tracer provider as we defined and registered above. 
+A tracer requires an instrumentation name, and an optional version to be created:
+```swift
```
Comment (svrnm) [inline content/en/docs/instrumentation/swift/manual.md:61]:
```suggestion
A tracer requires an instrumentation name, and an optional version to be created:

```swift
```

File: content/en/docs/instrumentation/swift/manual.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+let span =  let builder = tracer.spanBuilder(spanName: "\(name)").startSpan()
+...
+span.end()
+```
+It is required to call `end()` to end the span.
+
+### Creating Nested Spans
+
+Often, spans are used to build relationship between operations.
```
Comment (svrnm) [inline content/en/docs/instrumentation/swift/manual.md:79]:
```suggestion
Spans are used to build relationship between operations. 
```

File: content/en/docs/instrumentation/swift/manual.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+  // do work
+  childSpan.end()
+}
+
+```
+
+### Getting the Current Span
+
+Sometimes it's useful to do something with the current/active span. Here's how to access the current span from an arbitrary point in your code.
+```swift
```
Comment (svrnm) [inline content/en/docs/instrumentation/swift/manual.md:124]:
```suggestion
Sometimes it's useful to do something with the current/active span. Here's how to access the current span from an arbitrary point in your code.

```swift 
```

File: content/en/docs/instrumentation/swift/manual.md [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+Sometimes it's useful to do something with the current/active span. Here's how to access the current span from an arbitrary point in your code.
+```swift 
+  let currentSpan = OpenTelemetry.instance.contextProvider.activeSpan
+```
+### Span Attributes
+
+Spans can also be annotated with additional attributes. All spans will be automatically annotated with the `Resource` attributes attached to the tracer provider.
+The Opentelementry-swift sdk already provides instrumentation of common attributes in the `SDKResourceExtension` instrumentation.
+In this example a span for a network request capturing details about that request using existing [semantic conventions](https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/).
+```swift
```
Comment (svrnm) [inline content/en/docs/instrumentation/swift/manual.md:132]:
```suggestion
In this example a span for a network request capturing details about that request using existing [semantic conventions](https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/).

```swift
```

File: content/en/docs/instrumentation/swift/manual.md [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+            span.addEvent(name: "computation complete", attributes: attributes)
+```
+
+### Setting Span Status
+  
+### Recording Errors in Spans
+
+## Acquiring A Meter
+
+
```
Comment (svrnm) [inline content/en/docs/instrumentation/swift/manual.md:155]:
```suggestion
```

We can add those additional sections also in a follow-up PR, no problem in building this gradually.

**Review by svrnm — CHANGES_REQUESTED** [review]

**Comment by svrnm** [comment]
@bryce-b thanks! Looks like a good start to me

File: content/en/docs/instrumentation/swift/manual.md [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+linkTitle: Manual
+weight: 5
+description: Manual instrumentation for opentelemetry-swift
+---
+
+## Setup
+
+The [OpenTelemetrySdk](https://github.com/open-telemetry/opentelemetry-swift/blob/main/Sources/OpenTelemetrySdk/OpenTelemetrySdk.swift#L37) provides limited functionality in its default configuration. For more useful functionality, some configuration is required. 
+
+The default registered `TracerProvider` and `MetricProvider` are not configured with an exported. There are several [exporters](https://github.com/open-telemetry/opentelemetry-swift/tree/main/Sources/Exporters) available depending on your needs. Below we will explore configuring the OTLP exporter, which can be used for sending data to the [opentelemetry-collector](https://github.com/open-telemetry/opentelemetry-collector).
```
Comment (nachoBonafonte) [inline content/en/docs/instrumentation/swift/manual.md:12]:
```suggestion
The default registered `TracerProvider` and `MetricProvider` are not configured with an exporter. There are several [exporters](https://github.com/open-telemetry/opentelemetry-swift/tree/main/Sources/Exporters) available depending on your needs. Below we will explore configuring the OTLP exporter, which can be used for sending data to the [opentelemetry-collector](https://github.com/open-telemetry/opentelemetry-collector).
```

**Review by nachoBonafonte — APPROVED** [review]
Looks really good and clear to me. Thanks a lot for the effort.

**Comment by svrnm** [comment]
@bryce-b, can you make sure that the website builds in the tests:

```
10:45:43 PM: ERROR 2022/10/17 20:45:43 docs/instrumentation/swift/manual.md: use a local path, not an external URL, for the following reference to a local page: https://opentelemetry.io/docs/concepts/signals/traces/#spans-in-opentelemetry
10:45:43 PM: ERROR 2022/10/17 20:45:43 docs/instrumentation/swift/manual.md: use a local path, not an external URL, for the following reference to a local page: https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/
10:45:43 PM: ERROR 2022/10/17 20:45:43 docs/instrumentation/swift/manual.md: use a local path, not an external URL, for the following reference to a local page: https://opentelemetry.io/docs/concepts/signals/traces/#span-status
10:45:45 PM: Error: Error building site: logged 3 error(s)
```

File: content/en/docs/instrumentation/swift/manual.md [hunk 9]
```diff
@@ -0,0 +1,248 @@
...
+  span.addEvent(name: SemanticAttributes.exception.rawValue,
+    attributes: [SemanticAttributes.exceptionType.rawValue: AttributeValue.string(String(describing: type(of: error))),
+                 SemanticAttributes.exceptionEscaped.rawValue: AttributeValue.bool(false),
+                 SemanticAttributes.exceptionMessage.rawValue: AttributeValue.string(error.localizedDescription)])
+  })
+  span.status = .error(description: error.localizedDescription)
+}
+span.end()
+```
+## Metrics
```
Comment (svrnm) [inline content/en/docs/instrumentation/swift/manual.md:182]: We should skip this section until metrics are good to go.

File: content/en/docs/instrumentation/swift/manual.md (hunk 2 above)
Comment (cartermp) [reply content/en/docs/instrumentation/swift/manual.md]: Yes, I agree. We can add more and more of this later.

File: content/en/docs/instrumentation/swift/manual.md (hunk 9 above)
Comment (cartermp) [reply content/en/docs/instrumentation/swift/manual.md:182]: Yes, agreed

File: content/en/docs/instrumentation/swift/manual.md [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+
+```swift
+func parent() {
+  let parentSpan = someTracer.spanBuilder(spanName: "parent span").startSpan() 
+  child(span: parentSpan) 
+  parentSpan.end()
+}
+
+func child(parentSpan: Span) {
+  let childSpan = someTracer.spanBuilder(spanName: "child span")
```
Comment (svrnm) [inline content/en/docs/instrumentation/swift/manual.md:89]:
```suggestion
let childSpan = someTracer.spanBuilder(spanName: "child span")
```

File: content/en/docs/instrumentation/swift/manual.md [hunk 11]
```diff
@@ -0,0 +1,217 @@
...
+  }
+  guard let criticalData = get() else {
+      span.status = .error(description: "something bad happened")
+      return 
+  }
+  // do somthing 
+}
+```
+### Recording exceptions in Spans
+Semantic conventions provide special demarcation for events that record exceptions:
```
Comment (svrnm) [inline content/en/docs/instrumentation/swift/manual.md:167]:
```suggestion

### Recording exceptions in Spans
Semantic conventions provide special demarcation for events that record exceptions:
```

Comment (svrnm) [reply content/en/docs/instrumentation/swift/manual.md:167]: add a new line before the new section starts

File: content/en/docs/instrumentation/swift/manual.md [hunk 12]
```diff
@@ -0,0 +1,217 @@
...
+```swift
+  let span = someTracer.spanBuilder(spanName: "my span").startSpan()
```
Comment (svrnm) [inline content/en/docs/instrumentation/swift/manual.md:169]:
```suggestion
let span = someTracer.spanBuilder(spanName: "my span").startSpan() 
```

**Review by svrnm — APPROVED** [review]
minor things, nothing that should stop us, lgtm.

**Comment by svrnm** [comment]
https://opentelemetry.io/docs/instrumentation/swift/manual/ 🎉 

thanks @bryce-b

**Pull request merged**